SQLExecutorComponent/src/main/java/com/astronomvm/sqlExecutor/SQLExecutorComponent.java

```diff
@@ -0,0 +1,110 @@
+import com.astronomvm.core.model.data.row.AstronomObject;
+import com.astronomvm.core.model.data.row.Column;
+import com.astronomvm.core.model.data.row.Row;
+import com.astronomvm.core.model.data.row.RowHeader;
+import com.astronomvm.core.model.data.type.DataType;
+import com.astronomvm.core.model.database.DatabaseConnectionProperties;
+import com.astronomvm.core.model.database.JDBCDriver;
+import com.astronomvm.core.model.meta.ComponentMeta;
+import com.astronomvm.core.model.meta.ParameterMeta;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONObject;
+
+import java.sql.*;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+@Slf4j
+public class SQLExecutorComponent extends AstronomBaseComponent {
+
+    private static final String SQL_QUERY_PARAMETER_NAME = "SQL_QUERY";
+    private static final String DATABASE_CONNECTION_PROPERTIES_PARAMETER_NAME = "DATABASE_CONNECTION_PROPERTIES";
+
+
+    private String sqlQuery;
+    private DatabaseConnectionProperties dcp;
+
+
+    @Override
+    public ComponentMeta getComponentMeta() {
+        ComponentMeta componentMeta = new ComponentMeta();
+        componentMeta.setName("SQL_EXECUTOR");
+
+        ParameterMeta sqlQueryParameterMeta = new ParameterMeta();
+        sqlQueryParameterMeta.setName(SQL_QUERY_PARAMETER_NAME);
+        sqlQueryParameterMeta.setType(DataType.STRING);
+
+        ParameterMeta databaseConnectionPropertiesParameterMeta = new ParameterMeta();
+        databaseConnectionPropertiesParameterMeta.setName(DATABASE_CONNECTION_PROPERTIES_PARAMETER_NAME);
+        databaseConnectionPropertiesParameterMeta.setType(DataType.DATABASE_CONNECTION_PROPERTIES);
+
+        componentMeta.addParameterMeta(sqlQueryParameterMeta);
+        componentMeta.addParameterMeta(databaseConnectionPropertiesParameterMeta);
+
+        return componentMeta;
+    }
+
+    @Override
+    public void parseInputParameters(Map<String, JSONObject> parametersValues) {
+        this.sqlQuery = parametersValues.get(SQL_QUERY_PARAMETER_NAME).getString("value");
+        this.dcp = new DatabaseConnectionProperties();
+        JSONObject dcpJson = parametersValues.get(DATABASE_CONNECTION_PROPERTIES_PARAMETER_NAME).getJSONObject("value");
+        this.dcp.setDatabaseUrl(dcpJson.getString("databaseUrl"));
+        this.dcp.setJdbcDriver(JDBCDriver.valueOf(dcpJson.getString("jdbcDriver")));
+        this.dcp.setPassword(dcpJson.getString("password"));
+        this.dcp.setUsername(dcpJson.getString("username"));
+        this.dcp.setSchemaName(dcpJson.getString("schemaName"));
+    }
+
+    @Override
+    public ResultFlow execute() throws ComponentException {
+        ResultFlow resultFlow = new ResultFlow();
+
+        try {
+            Class.forName(this.dcp.getJdbcDriver().getClassName());
+            Connection connection = DriverManager.getConnection(this.dcp.getDatabaseUrl(),this.dcp.getUsername(),this.dcp.getPassword());
+            Statement statement = connection.createStatement();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use try-with-resources or close this "Statement" in a "finally" clause. ([squid:S2095](https://sonarcloud.io/organizations/ahmedriahi-github/rules#rule_key=squid%3AS2095))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=AhmedRiahi_AstronomVM&pullRequest=22&issues=AWZfQr15DZywvPe4LfuL&open=AWZfQr15DZywvPe4LfuL)

[ISSUE_KEY]: AWZfQr15DZywvPe4LfuL